common/crypto/sanitize.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package crypto
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"math/big"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/hyperledger/fabric/bccsp/utils"
+	"github.com/pkg/errors"
+)
+
+// SanitizeIdentity sanitizes the signature scheme of the identity
+func SanitizeIdentity(identity []byte) ([]byte, error) {
+	sID := &msp.SerializedIdentity{}
+	if err := proto.Unmarshal(identity, sID); err != nil {
+		return nil, errors.Wrapf(err, "failed unmarshaling identity %s", string(identity))
+	}
+
+	der, _ := pem.Decode(sID.IdBytes)
+	if der == nil {
+		return nil, errors.Errorf("failed to PEM decode identity bytes: %s", string(sID.IdBytes))
+	}
+	cert, err := x509.ParseCertificate(der.Bytes)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed parsing certificate %s", string(sID.IdBytes))
+	}
+
+	r, s, err := utils.UnmarshalECDSASignature(cert.Signature)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed unmarshaling ECDSA signature on identity: %s", string(sID.IdBytes))
+	}
+
+	// We assume that the consenter and the CA use the same signature scheme.
+	curveOrderUsedByCryptoGen := cert.PublicKey.(*ecdsa.PublicKey).Curve.Params().N
```

Comment:
This code will fail if the certificate is not signed by ECDSA. 

Let's proceed but open an issue for this.

Reply:
https://github.com/hyperledger/fabric/issues/4007